R/data.R

```diff
@@ -482,6 +501,60 @@ data_path_request <- function(path_data) {
   file.path(data_path_cyphr(path_data), "requests")
 }
 
+data_path_user_key <- function(path_data, hash, version) {
+  path <- data_path_cyphr(path_data)
+  if (version >= numeric_version("1.1.0")) {
+    path <- file.path(path, "keys")
+  }
+  if (is.null(hash)) {
+    path
+  } else {
+    file.path(path, bin2str(hash, ""))
+  }
+}
+
+data_path_version <- function(path_data) {
+  file.path(data_path_cyphr(path_data), "version")
+}
+
+data_version_write <- function(path_data) {
+  dest <- data_path_version(path_data)
+  dir.create(dirname(dest), FALSE, TRUE)
+  writeLines(as.character(data_schema_version()),
+             data_path_version(path_data))
+}
+
+data_version_read <- function(path_data) {
+  path_data <- normalizePath(path_data, mustWork = TRUE)
+  if (!exists(path_data, data_cache$versions)) {
+    p <- data_path_version(path_data)
+    if (file.exists(p)) {
+      v <- numeric_version(readLines(p))
+    } else {
+      v <- numeric_version("1.0.0")
+    }
+    cur <- data_schema_version()
+    if (v < cur) {
+      msg <- c(
+        sprintf(
+          "Your cyphr schema version is out of date (found %s, current is %s)",
+          v, cur),
+        sprintf(
+          'Please run cyphr:::data_schema_migrate("%s")',
```

Comment:
Any reason to have this a non-exported function? Is that just because you're not expecting migration to be necessary?

Reply:
yes, precisely :) - and if it's not exported we can think about the interface more if we need to update in the future